module/state_synchronization/indexer/persister.go

```diff
@@ -0,0 +1,221 @@
+package indexer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+	"go.uber.org/multierr"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/store/inmemory/unsynchronized"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+// Persister handles transferring data from in-memory storages to permanent storages.
+type Persister struct {
```

Comment:
Is the Persister a singleton or a Persister per height?

Looks like the header and executionResult are not updatable, so I assume the persister is only for a single block?

If that's the case, we need to make it clear. Maybe even rename it into BlockPersister, indicating this is only for a block.

Also, is `inMemoryRegisters` shared by different blocks or just for this block?